### DISCUSSION TITLE
Given Step definition can be re-used as a Then

### DISCUSSION THREAD

**Discussion by abiencourt** [badeball/cypress-cucumber-preprocessor#1205] [General]
It came to my attention that I am able to use a `Given` step definition as a `Then` in my feature file (see example below with "I navigate to"). This run successfully and does not print any warning or being flagged anywhere during the test run.

Is this expected behavior? I was not expecting a `Then` step capable of using a `Given` step definition. Could you please shed some light on this?

Thanks!

---

Versions:
@badeball/cypress-cucumber-preprocessor: 16.0.3
@bahmutov/cypress-esbuild-preprocessor: 2.1.0
cypress: 12.17.4

```cucumber
    Scenario: Remove hotel from Competitive Set
        Given I can see "Competitive List"
        When I "uncheck" the 'Show' checkbox for hotel "hotel1"
        And I click on "Save Changes"
        Then I navigate to "1021/yield-sheet"
        And I see "hotel1"

    Scenario: Verify navigation from clicking on the logo
        Given I navigate to "/test"
        When I click on image ID "navbar-logo"
        Then I am redirected to "/"
        And I see "Hotel1"
```

```ts
import { Given } from "@badeball/cypress-cucumber-preprocessor";

Given("I navigate to {string}", (path: string) => {
  cy.visit(path, { timeout: 10000 });
});
```

**Comment by badeball** [comment]
> Is this expected behavior?

Yes, this is expected behavior and how `cucumber-js` behaves.

**Reply by abiencourt** [reply]
Thanks for confirming. I kept digging after posting this question and found these references:
- [Cucumber-js unable to handle Given and When steps with same step text](https://github.com/cucumber/cucumber-js/issues/1696)
- [Keywords : Strict keyword enforcement](https://github.com/cucumber/common/issues/1392)

**Reply by badeball** [reply]
Yup, this is a [proposed change](https://github.com/cucumber/common/issues/768). It looks like it has made its way into the Go implementation, while some progress has been made on the JS implementation.
